app/controllers/assessments_controller.rb

```diff
@@ -185,6 +180,7 @@ def importAsmtFromTar
         # Ensure file will lie within course, otherwise skip
         # Allow equality for the main directory to be created
         next unless Archive.in_dir?(Pathname(entry_file), Pathname(assessment_path), strict: false)
+        next if !existing_asmt.nil? && relative_pathname.start_with?(existing_asmt.handin_directory)
```

Comment:
This check doesn't actually work, because `relative_pathname` is prepended by the assessment's name, e.g. `homework3/handin/`

Reply:
I would suggest replacing the check with something like

```ruby
next if existing_asmt && Archive.in_dir?(Pathname(entry_file), existing_asmt.handin_directory_path, strict: false)
```

Reply:
Verify that the check works by adding a file to the handin directory, and upload a tar where that file is replaced with something else -> shouldn't overwrite